.github/actions/buildAndPush/action.yml

```diff
@@ -17,7 +26,16 @@ runs:
     - name: Set Up Docker to Use gcloud Credentials
       run: gcloud auth configure-docker -q
       shell: bash
-    - run: docker build -t ${{ inputs.image-path }} -f ${{ inputs.dockerfile }} .
+    # Note: it is not generally recommended to pass secrets as build args, but
```

Comment:
For the prod workflow, instead of building containers, they are copied to the prod project's repo. 

That won't work with this change. 

I'd like to keep copying at least the backend containers with the idea that the containers from the test environment have been verified and tested already. 

Should I change the prod workflow to build the frontend container with the new build arguments instead of just copying from test?

Reply:
Hmm. An alternative is to provide these as runtime environment variables. That would work fine for the username/password (and arguably it's better to do it that way because it avoid this not-recommended way of passing sensitive data to a container).

However, the build-context argument is needed at build-time to determine which environment file to use for the frontend. I don't think there's a good way around that unfortunately. We could in theory build support for runtime environment variables in the frontend but I don't think that's a good idea because it introduces a bunch of complexities like concerns around XSS, caching, etc.

I guess I'd say the frontend should just be built separately, unless you have another idea?

Reply:
I don't know enough about the js container runtime to have any other suggestions. Building the frontend for prod should work fine. Just want to make sure it's good with you.

Reply:
Yeah it's fine with me. The frontend container is simple enough, we'll just need to make sure that prod releases are validated immediately after pushing. But I think we'd need to do that anyway since environment variables would change the behavior regardless of whether they're passed at build time or runtime.

Reply:
Yep, the e2e tests will still be run right after the prod resources are updated.